src/west/app/project.py

```diff
@@ -602,6 +602,118 @@ def _dump(self, args, to_dump):
         else:
             sys.stdout.write(to_dump)
 
+class Compare(_ProjectCommand):
+    def __init__(self):
+        super().__init__(
+            'compare',
+            "compare project status against the manifest",
+            textwrap.dedent('''\
+            Compare each project's working tree state against the results
+            of the most recent successful "west update" command.
+
+            This command prints output for a project if (and only if)
+            at least one of the following is true:
+
+            1. its checked out commit is different than the commit checked
+               out by the most recent successful "west update"
+            2. its working tree is not clean (i.e. there are local uncommitted
+               changes)
+            3. it has a local checked out branch
+
+            The command prints output for the manifest repository if it
+            has nonempty status.
+
+            The output is meant to be human-readable, and may change. It is
+            not a stable interface to write scripts against.''')
+        )
+
+    def do_add_parser(self, parser_adder):
+        parser = self._parser(parser_adder,
+                              epilog=ACTIVE_CLONED_PROJECTS_HELP)
+        parser.add_argument('projects', metavar='PROJECT', nargs='*',
+                            help='''projects (by name or path) to operate on;
+                            defaults to active cloned projects''')
+        parser.add_argument('-a', '--all', action='store_true',
+                            help='include output for inactive projects')
+        parser.add_argument('--exit-code', action='store_true',
+                            help='''exit with status code 1 if status output
+                            was printed for any project''')
+        parser.add_argument('--ignore-branches', action='store_true',
+                            help='''skip output for projects with checked out
+                            branches and clean working trees if the branch is
+                            at the same commit as the last "west update"''')
+        return parser
+
+    def do_run(self, args, ignored):
+        self.die_if_no_git()
+        if self.git_version_info < (2, 22):
+            self.die('git version 2.22 or later is required')
```

Comment:
remember to add to release notes that git >=2.22 is now required if using this command.

Also I believe it would be nice with a comment as to why git >=2.22 is required.
I assume it's to due to the `branch --show-current` which were introduced in 2.22.

But thanks for only having this as requirement for this command 👍 and not enforce it generally for `west`.